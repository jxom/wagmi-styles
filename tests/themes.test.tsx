import React from 'react';
import { View } from 'react-native';
import { render } from '@testing-library/react-native';

import * as FooTheme from './theme';
import * as BarTheme from './theme2';

describe('themed', () => {
  const simpleStyleRef = FooTheme.style({
    height: '100%',
    alignItems: 'center',
    justifyContent: 'center',
    display: 'flex',
  });

  function Simple() {
    const simpleStyle = FooTheme.useStyle(simpleStyleRef);
    return <View style={simpleStyle} />;
  }

  test('simple style (wo/ ThemeProvider)', () => {
    const { toJSON } = render(<Simple />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('simple style (w/ ThemeProvider)', () => {
    const { toJSON } = render(
      <FooTheme.ThemeProvider theme={FooTheme.themeA}>
        <Simple />
      </FooTheme.ThemeProvider>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  ///////////////////////////////////////////////////////////////////////////

  const styleWithVarsRef = FooTheme.style((vars) => ({
    color: vars.colors.primary,
    backgroundColor: vars.colors.secondary,
    height: '100%',
    alignItems: 'center',
    justifyContent: 'center',
    display: 'flex',
  }));

  function StyleWithVars() {
    const styleWithVars = FooTheme.useStyle(styleWithVarsRef);
    return <View style={styleWithVars} />;
  }

  [FooTheme.themeA, FooTheme.themeB].forEach((theme) => {
    test('style with theme vars', () => {
      const { toJSON } = render(
        <FooTheme.ThemeProvider theme={theme}>
          <StyleWithVars />
        </FooTheme.ThemeProvider>
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });

  ///////////////////////////////////////////////////////////////////////////

  const styleWithVars2Ref = BarTheme.style((vars) => ({
    color: vars.colors.primary,
    backgroundColor: vars.colors.secondary,
    padding: vars.spacings.large,
    height: '100%',
    alignItems: 'center',
    justifyContent: 'center',
    display: 'flex',
  }));

  function StyleWithVars2() {
    const styleWithVars2 = BarTheme.useStyle(styleWithVars2Ref);
    return <View style={styleWithVars2} />;
  }

  test('nested themes', () => {
    const { toJSON } = render(
      <FooTheme.ThemeProvider theme={FooTheme.themeA}>
        <StyleWithVars />
        <FooTheme.ThemeProvider theme={FooTheme.themeB}>
          <StyleWithVars />
          <BarTheme.ThemeProvider theme={BarTheme.themeC}>
            <StyleWithVars />
            <StyleWithVars2 />
          </BarTheme.ThemeProvider>
        </FooTheme.ThemeProvider>
      </FooTheme.ThemeProvider>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
