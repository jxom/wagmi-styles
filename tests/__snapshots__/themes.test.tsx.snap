// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themed nested themes 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "hotpink",
          "color": "blue",
          "display": "flex",
          "height": "100%",
          "justifyContent": "center",
        },
      ]
    }
  />,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "black",
          "color": "red",
          "display": "flex",
          "height": "100%",
          "justifyContent": "center",
        },
      ]
    }
  />,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "black",
          "color": "red",
          "display": "flex",
          "height": "100%",
          "justifyContent": "center",
        },
      ]
    }
  />,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "yellow",
          "color": "green",
          "display": "flex",
          "height": "100%",
          "justifyContent": "center",
          "padding": 16,
        },
      ]
    }
  />,
]
`;

exports[`themed simple style 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
      },
    ]
  }
/>
`;

exports[`themed style with theme vars 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "hotpink",
        "color": "blue",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
      },
    ]
  }
/>
`;

exports[`themed style with theme vars 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "color": "red",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
      },
    ]
  }
/>
`;
